### ISSUE TITLE
Splits should keep group examples together

### ISSUE THREAD

**Issue by gsganden** [ai-evals-course/recipe-chatbot#36]
https://github.com/ai-evals-course/recipe-chatbot/blob/main/homeworks/hw3/data/labeled_traces.csv contains multiple examples for each query. Putting some examples from a given query e.g. in the training set and some in the test set is a source of leakage. https://github.com/ai-evals-course/recipe-chatbot/blob/main/homeworks/hw3/README.md does not mention this issue, nor does https://github.com/ai-evals-course/recipe-chatbot/blob/main/homeworks/hw3/hw3_walkthrough.ipynb.

This point does add complexity and is somewhat hard to explain, but I suggest that it's important enough to be worth covering.
